File: frontend/src/types/project-metadata.ts
export interface ProjectStatistics {
  total_files: number;
  total_folders: number;
  total_size_bytes: number;
  source_files: number;
  config_files_count: number;
  documentation_files: number;
  image_files: number;
  video_files: number;
  asset_files: number;
}

export interface DevOpsInfo {
  docker: boolean;
  docker_compose: boolean;
  kubernetes: boolean;
  ci_cd: string[];
}

export interface ProjectMetadata {
  project_name: string;
  project_type: string | null;
  primary_language: string | null;
  secondary_languages: string[];
  languages: Record<string, number>;
  frameworks: string[];
  package_manager: string | null;
  databases: string[];
  devops: DevOpsInfo;
  config_files: string[];
  has_readme: boolean;
  statistics: ProjectStatistics;
  last_scanned_at: string | null;
}

File: backend/app/schemas/project_metadata.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class ProjectStatistics(BaseModel):
    total_files: int
    total_folders: int
    total_size_bytes: int
    source_files: int
    config_files_count: int
    documentation_files: int
    image_files: int
    video_files: int
    asset_files: int


class DevOpsInfo(BaseModel):
    docker: bool
    docker_compose: bool
    kubernetes: bool
    ci_cd: list[str]


class ProjectMetadataResponse(BaseModel):
    project_name: str
    project_type: Optional[str] = None
    primary_language: Optional[str] = None
    secondary_languages: list[str]
    languages: dict[str, int]
    frameworks: list[str]
    package_manager: Optional[str] = None
    databases: list[str]
    devops: DevOpsInfo
    config_files: list[str]
    has_readme: bool
    statistics: ProjectStatistics
    last_scanned_at: Optional[datetime] = None

    model_config = {"from_attributes": True}
